Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialComponentSlot.cpp

```diff
@@ -219,37 +216,43 @@ namespace AZ
             }
         }
 
-        void EditorMaterialComponentSlot::OpenMaterialInspector()
+        void EditorMaterialComponentSlot::OpenMaterialEditor() const
         {
-            MaterialPropertyOverrideMap initialPropertyOverrides = m_propertyOverrides;
-            auto applyPropertyChangedCallback = [this](const MaterialPropertyOverrideMap& propertyOverrides) {
-                m_propertyOverrides = propertyOverrides;
-                OnPropertyChanged();
-            };
-
-            if (m_materialAsset.GetId().IsValid())
+            const AZStd::string& sourcePath = AZ::RPI::AssetUtils::GetSourcePathByAssetId(m_materialAsset.GetId());
+            if (!sourcePath.empty() && AZ::StringFunc::Path::IsExtension(sourcePath.c_str(), AZ::RPI::MaterialSourceData::Extension))
```

Comment:
may make sense to AZ_Error if source path is not what's expected

Reply:
Since we support passing in an empty string I'm not sure what we would hear on unless we were checking for the file's existence. But the material editor itself would display an error about that.